hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadIncrementalRelation.scala

```diff
@@ -122,6 +144,17 @@ class MergeOnReadIncrementalRelation(sqlContext: SQLContext,
     }
     filteredFileSlices
   }
+
+  private def setUpFilterParams(requiredSchema: HoodieTableSchema): FilterParams = {
+    val attrs = requiredSchema.structTypeSchema.map(f => AttributeReference(f.name, f.dataType, f.nullable, f.metadata)())
+    val filterExpression = convertToExpressions(incrementalSpanRecordFilters.toArray).map { e =>
+      e transform {
+        case a: AttributeReference =>
+          attrs.find(_.name.equalsIgnoreCase(a.name)).get
```

Comment:
Instead of manipulating expressions like that we should instead fix `convertToExpressions` by:
 - Making it produce unresolved expressions, replacing name-refs w/ `UnresolvedAttribute` instead of `AttributeReference` during conversion
 - Then resolve converted expression using `resolveExpr` (in the same utility)